pmd-apex/src/test/resources/net/sourceforge/pmd/lang/apex/rule/performance/xml/OperationWithHighCostInLoop.xml

```diff
@@ -0,0 +1,48 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<test-data
+    xmlns="http://pmd.sourceforge.net/rule-tests"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="http://pmd.sourceforge.net/rule-tests http://pmd.sourceforge.net/rule-tests_1_0_0.xsd">
+
+    <!-- Begin Schema method invocations -->
+    <test-code>
+        <description>High cost performance getGlobalDescribe in loop</description>
+        <expected-problems>1</expected-problems>
+        <code>
+            <![CDATA[
+            public class Foo {
+                public void testGetGlobalDescribe() {
+                    Set<String> fieldNameSet = new Set<String> {'Id'};
+                    for (String fieldNameOrDefaultValue : fieldNameOrDefaultValueList) {
+                        if (Schema.getGlobalDescribe().get(objectName).getDescribe().fields.getMap().containsKey(fieldNameOrDefaultValue.trim() )) {
+                            fieldNameSet.add(fieldNameOrDefaultValue);
+                        }
+                    }
+                }
+            }
+                    ]]>
+        </code>
+    </test-code>
+
+    <test-code>
+        <description>High cost performance describeSObjects in loop</description>
```

Comment:
There are two Schema.describeSObjects overloads - please also add a test case for the second variant: `describeSObjects(SObjectTypes, SObjectDescribeOptions)`

Reply:
Additionally, there might be a fully qualified calls, like `System.Schema.getGlobalDescibe()` and `System.Schema.describeSObjects(...)` which should be covered in test cases